metrics/timer_test.go

```diff
@@ -0,0 +1,33 @@
+package metrics_test
+
+import (
+	"math"
+	"testing"
+
+	"time"
+
+	"github.com/go-kit/kit/metrics"
+	"github.com/go-kit/kit/metrics/generic"
+)
+
+func TestTimerFast(t *testing.T) {
+	h := generic.NewSimpleHistogram()
+	metrics.NewTimer(h).ObserveDuration()
+
+	tolerance := 0.005
```

Comment:
We've seen 10ms not be enough in timing sensitive Prometheus tests. I'd use at least 50ms.